File: backend/app/schemas/notification_logs.py
from pydantic import BaseModel
from datetime import datetime

class NotificationLogCreate(BaseModel):
    schedule_id: int
    status: str
    message: str | None = None

class NotificationLogResponse(BaseModel):
    id: int
    schedule_id: int
    status: str
    message: str | None = None
    sent_at: datetime 

    class Config:
        from_attributes = True
    

File: frontend/src/api/notification_logs.ts
import api from "./axios";

export interface NotificationLogResponse {
    id: number;
    schedule_id: number;
    status: string;
    message: string;
    sent_at: string;
}


export const getNotificationLogs = async(): Promise<NotificationLogResponse[]> => {
    const response = await api.get<NotificationLogResponse[]>('/notification_logs');
    return response.data;
}